### ISSUE TITLE
AvroWrite exponentional file size

### ISSUE THREAD

**Issue by zoltan-fedor** [mtth/hdfs#31]
I have found a very interesting issue with AvroWriter, which breaks the whole writing process.
Basically the files size start growing exponentially.
When I export 100 records, the file size is 10Kb, when I export 200 records (of the same), the file size is 670Kb!!!

I use the standard AvroWriter example (except added a long string value to make the data bigger).
How to reproduce:
The below code exports 2 records. just take those 2 records and copy-paste them until you don't have 200 records of the same and then run the export. Observe that the file size became 670Kb.
Remove half of the records, leave only 100 records. Export again, observe that the file size is 10Kb!

```
"""Avro extension example."""
from hdfs import InsecureClient
from hdfs.ext.avro import AvroReader, AvroWriter
import os

os.environ['http_proxy'] = ''
os.environ['user'] = 'my_user'

# Get the default alias' client.
client = InsecureClient('http://xxx:50070', user='my_user')

# Some sample data.
records = [
    {'name': 'Ann', 'age': 23, 'string': 'asdasd asklhj doiuiroe t43785 iofdjkvnuhyt789u judffkguirety9 jdfvhjkfcvhuierytuiy348957hjdfh jkdfgirhytiery5689347589'},
    {'name': 'Bob', 'age': 22, 'string': 'nkdfn kjfhy tuier vgfhe89734 89hIUHIUJKBNUIh iuhiuh48 9 5u7489jkldjvlcdjvoiejuioweufoksdfsdf'},
]

# Write an Avro File to HDFS (since our records' schema is very simple, we let
# the writer infer it automatically, otherwise we would pass it as argument).
# codec can be null, deflate or snappy
with AvroWriter(client, '/user/my_user/names.avro', overwrite=True) as writer:
    for record in records:
        writer.write(record)
```

**Comment by mtth** [comment]
Thanks for the detailed info. Which version of HdfsCLI are you on?
I wasn't able to reproduce this behavior with your example on `2.0.2` (the size increases linearly instead).

**Comment by zoltan-fedor** [comment]
That's interesting, because I am on the latest version too, 2.0.2 on Python 3.4.3.

**Comment by mtth** [comment]
Let me try on python 3 then.

**Comment by mtth** [comment]
Great, I can reproduce it! Will look into it tonight.

**Comment by zoltan-fedor** [comment]
Great! Thanks for looking into it!!!

**Comment by mtth** [comment]
Should be fixed in 3b09fb310141e689b14c1273f5544c737a536bf2, just released version `2.0.3` with the fix. Thanks for the great bug report!

**Comment by zoltan-fedor** [comment]
Thanks for the quick fix!
I have loaded the new version and I can confirm, that it has solved the problem.
Thanks again.
